store/postgres/src/relational_queries.rs

```diff
@@ -1205,6 +1207,96 @@ impl<'a> LoadQuery<PgConnection, EntityData> for FindQuery<'a> {
 
 impl<'a, Conn> RunQueryDsl<Conn> for FindQuery<'a> {}
 
+/// Builds a query over a given set of [`Table`]s in an attempt to find updated
+/// and/or newly inserted entities at a given block number; i.e. such that the
+/// block range's lower bound is equal to said block number.
+#[derive(Debug, Clone, Constructor)]
+pub struct FindChangesQuery<'a> {
+    pub(crate) _namespace: &'a Namespace,
+    pub(crate) tables: &'a [&'a Table],
+    pub(crate) block: BlockNumber,
+}
+
+impl<'a> QueryFragment<Pg> for FindChangesQuery<'a> {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+
+        for (i, table) in self.tables.iter().enumerate() {
+            if i > 0 {
+                out.push_sql("\nunion all\n");
+            }
+            out.push_sql("select ");
+            out.push_bind_param::<Text, _>(&table.object.as_str())?;
+            out.push_sql(" as entity, to_jsonb(e.*) as data\n");
+            out.push_sql("  from ");
+            out.push_sql(table.qualified_name.as_str());
+            out.push_sql(" e\n where ");
+            BlockRangeLowerBoundClause::new("e.", self.block).walk_ast(out.reborrow())?;
+        }
+
+        Ok(())
+    }
+}
+
+impl<'a> QueryId for FindChangesQuery<'a> {
+    type QueryId = ();
+
+    const HAS_STATIC_QUERY_ID: bool = false;
+}
+
+impl<'a> LoadQuery<PgConnection, EntityData> for FindChangesQuery<'a> {
+    fn internal_load(self, conn: &PgConnection) -> QueryResult<Vec<EntityData>> {
+        conn.query_by_name(&self)
+    }
+}
+
+impl<'a, Conn> RunQueryDsl<Conn> for FindChangesQuery<'a> {}
+
+/// Builds a query over a given set of [`Table`]s in an attempt to find deleted
+/// entities; i.e. such that the block range's lower bound is equal to said
+/// block number.
+#[derive(Debug, Clone, Constructor)]
+pub struct FindDeletesQuery<'a> {
+    pub(crate) _namespace: &'a Namespace,
+    pub(crate) tables: &'a [&'a Table],
+    pub(crate) block: BlockNumber,
+}
+
+impl<'a> QueryFragment<Pg> for FindDeletesQuery<'a> {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+
+        for (i, table) in self.tables.iter().enumerate() {
+            if i > 0 {
+                out.push_sql("\nunion all\n");
+            }
+            out.push_sql("select ");
+            out.push_bind_param::<Text, _>(&table.object.as_str())?;
+            out.push_sql(" as entity, to_jsonb(e.*) as data\n");
```

Comment:
It would be better for performance to only select the id here instead of the entire object, i.e., `select 'EntityType' as entity, e.id from ..` That would require doing the whole `QueryableByName` dance for a new struct, but is not  hard for reading two strings. Just adding this struct should do the trick:
```rust
#[derive(QueryableByName)]
struct EntityDeletion {
    #[sql_type = "Text"]
    entity: String,
    #[sql_type = "Text"]
    id: String,
}
```

Reply:
Makes sense, let's `select` as little data as possible. I wanted to reuse types we already had and compromise on query performance here, but I guess it's a good idea.

Reply:
Fixed.